libp2p/HostCapability.h

```diff
@@ -64,7 +64,7 @@ class HostCapability: public HostCapabilityFace
     std::shared_ptr<Capability> newPeerCapability(
         std::shared_ptr<SessionFace> const& _s, unsigned _idOffset, CapDesc const& _cap) override
     {
-        auto p = std::make_shared<PeerCap>(_s, this, _idOffset, _cap);
+        auto p = std::make_shared<PeerCap>(_s, name(), messageCount(), _idOffset, _cap);
```

Comment:
Why the capability `PeerCap` must own the session?

Reply:
It doesn't own, it stores a weak_ptr `Capability::m_session`

Reply:
Yeah I should change the constructor to take weak_ptr, too

Reply:
It's not obvious here if `_s` is passed by reference.

Reply:
Are you going to change it here?